Documentation/CRDs/ceph-client-crd.md

```diff
@@ -34,10 +38,121 @@ metadata:
   namespace: rook-ceph
 spec:
   caps:
-    mon: 'profile rbd'
+    mon: 'profile rbd, allow r'
     osd: 'profile rbd pool=volumes, profile rbd pool=vms, profile rbd-read-only pool=images'
 ```
 
-### Prerequisites
+To use `CephClient` to connect to a Ceph cluster:
 
-This guide assumes you have created a Rook cluster as explained in the main [Quickstart guide](../Getting-Started/quickstart.md).
+### 2. Find the generated secret for the `CephClient`
+
+Once your `CephClient` has been processed by Rook, it will be updated to include your secret:
+
+```console
+kubectl -n rook-ceph get cephclient example -o jsonpath='{.status.info.secretName}'
+```
+
+### 3. Extract Ceph cluster credentials from the generated secret
+
+Extract Ceph cluster credentials from the generated secret (note that the subkey will be your original client name):
+
+```console
+kubectl --namespace rook-ceph get secret rook-ceph-client-example -o jsonpath="{.data.example}" | base64 -d
+```
+
+The base64 encoded value that is returned *is* the password for your ceph client.
+
+### 4. Retrieve the mon endpoints
+
+To send writes to the cluster, you must retrieve the mons in use:
+
+```console
+kubectl --namespace rook-ceph get configmap rook-ceph-mon-endpoints -o jsonpath='{.data.data}' | sed 's/.=//g'`
+```
+
+This command *should* produce a line that looks somewhat like this:
+
+```
+10.107.72.122:6789,10.103.244.218:6789,10.99.33.227:6789
+```
+
+### 5. (optional) Generate Ceph configuration files
+
+If you choose to generate files for Ceph to use you will need to generate the following files:
+
+- General configuration file (ex. `ceph.conf`)
+- Keyring file (ex. `ceph.keyring`)
+- File containing ceph arguments to use (ex. `ceph.args`)
+
+Examples of the files follow:
+
+`ceph.conf`
+```ini
+[global]
+mon_host=10.107.72.122:6789,10.103.244.218:6789,10.99.33.227:6789
+log file = /tmp/ceph-$pid.log
+```
+
+`ceph.keyring`
+```ini
+[client.example]
+  key = < key, decoded from k8s secret>
+  # The caps below are for a rbd workload -- you may need to edit/modify these capabilities for other workloads
+  # see https://docs.ceph.com/en/latest/cephfs/capabilities
+  caps mon = 'allow r'
+  caps osd = 'profile rbd pool=<your pool>, profile rb pool=<another pool>'
+```
+
+`ceph.args`
```

Comment:
No need for this ceph.args file now if the env var doesn't use it anymore? Or if you really prefer it, you could keep it and revert my previous suggestion to have it inline with the env var.

Reply:
yeah I personally like the file, but since I went with inlining it I'll remove the file.

People can use a file and read contents if they find it easier